core/primitives/src/sharding.rs

```diff
@@ -399,17 +397,16 @@ impl ShardChunkHeader {
         }
     }
 
-    pub fn version_range(&self) -> ProtocolVersionRange {
-        let block_header_v3_version = ProtocolFeature::BlockHeaderV3.protocol_version();
+    /// Returns the range of `ProtocolVersion`s where this variant of the
+    /// message is accepted.  For the newest variant the upper limit is the
+    /// maximum protocol version.
+    pub fn version_range(&self) -> std::ops::Range<ProtocolVersion> {
+        const BLOCK_HEADER_V3_VERSION: ProtocolVersion =
+            ProtocolFeature::BlockHeaderV3.protocol_version();
```

Comment:
Wait, why do we need a `const` here? Just `let` would do. I don't think it's worth using `const` for local variables just to emphasize constness: 

* the optimizer is smart enough to constant-fold (if it wasn't, the whole iterator infrastructure would be horribly slow)
* SCREAMING_SNAKE_CASE looks jarring
* `const` is an item, it has very different name resolution rules from `let`

That being said, everything else here is CAPS already, so perhaps this makes code more uniform? No strong opinion either way.

Reply:
> That being said, everything else here is CAPS already, so perhaps this makes code more uniform?

Yeah, that’s pretty much my reasoning.  It IMO looks more aesthetic.